markdown/docs/addon-author-guide.md

```diff
@@ -0,0 +1,321 @@
+# Addon Author Guide
+
+If you're the author of an Ember addon, you might need to make some
+changes to ensure that your addon runs correctly in the FastBoot
+environment.
+
+Remember that when updating your addons for FastBoot compatibility,
```

Comment:
Adding similar language to the README (and site) for app developers. This is the most common class of errors everyone will encounter.